### ISSUE TITLE
Connecting & selecting in the same time fails

### ISSUE THREAD

**Issue by llfbandit** [vert-x3/vertx-redis-client#142]
### Version

* vert.x core: 3.7.0
* vert.x redis client: 3.7.0

### Context

Connecting with a pre-configured select option always fail.
This does not if with the selection is made after.

### Do you have a reproducer?
Here's a snippet to reproduce the behaviour :

```
var conf = new JsonObject("{
    "host": "localhost",
    "port": 6379,
    "select": 1
}");

Redis
  .createClient(vertx, new RedisOptions(conf))
  .connect(onConnect -> {
    // Failure : Future{cause=Redis connection is broken.}
  });
```

**Comment by akiyamayami** [comment]
Hi, Anyone can slive this, I just upgrade my project version vertx 3.5 to 3.7. but can not use Redis Client :(

**Comment by micean** [comment]
Because `RedisClient` set `connected = true` after send `select n` when `select` of RedisOptions is non-null.
Use the old client 'io.vertx.redis.RedisClient' before fix this bug

**Comment by pmlopes** [comment]
Fixed on master and 3.7.1
